lib/liquid/traversal.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module Liquid
+  class Traversal
+    def self.for(node, callbacks = Hash.new(proc {}))
+      kase = CASES.find { |(klass, _)| node.is_a?(klass) }&.last
```

Comment:
`CASES` isn't used for anything else, maybe it should be inlined to make the code clearer?

```rb
traversal_class = case node
when Liquid::Assign then Assign
when Liquid::Case then Case
...
```

Reply:
Doing it as an inline `case` expression fails policial because there are too many cases.  Doing it with data also makes it easier to extend this to a registry-style model in the future (maybe for custom liquid tags to add to or similar?)